sdks/typescript/client-typescript-core/test/repository_analytics_spec.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ClientFeatureRepository,
+  FeatureState,
+  FeatureStateHolder,
+  FeatureValueType,
+  SSEResultState
+} from '../app';
+import { expect } from 'chai';
+
+describe( 'We should be able to log an analytics event', () => {
+  let repo: ClientFeatureRepository;
+  let firedAction: string;
+  let firedOther: Map<string, string>;
+  let firedFeatures: Array<FeatureStateHolder>;
+
+  beforeEach(() => {
+    repo = new ClientFeatureRepository();
+    firedAction = undefined;
+    firedOther = undefined;
+    firedFeatures = undefined;
+
+    repo.addAnalyticCollector({
+      logEvent: function(action: string, other: Map<string, string>,
+                         featureStateAtCurrentTime: Array<FeatureStateHolder>) {
+        firedAction = action;
+        firedOther = other;
+        firedFeatures = featureStateAtCurrentTime;
+      }
+    });
+  });
+
+  it('Should enable us to log an event with no other and no features', () => {
+    repo.logAnalyticsEvent('name');
+    expect(firedFeatures.length).to.eq(0);
+    expect(firedAction).to.eq('name');
+    // tslint:disable-next-line:no-unused-expression
+    expect(firedOther).to.be.undefined;
+  });
+
+  it('should carry through the other field', () => {
```

Comment:
what are we testing here? just not sure of this description

Reply:
the field "other"